drizzle-orm/src/mysql-core/query-builders/select.types.ts

```diff
@@ -184,26 +197,51 @@ export interface MySqlSelectHKT extends MySqlSelectHKTBase {
 		Assume<this['nullabilityMap'], Record<string, JoinNullability>>,
 		this['dynamic'],
 		this['excludedMethods'],
-		this['result'],
+		Assume<this['result'], any[]>,
 		Assume<this['selectedFields'], ColumnsSelection>
 	>;
 }
 
+export type MySqlSetOperatorExcludedMethods = Exclude<
```

Comment:
2nd thing is, from what I saw in the docs, only Postgres allows using `limit`, `offset` etc. clauses on the result of the set operator query, i.e. `(select ...) union (select ...) limit X offset Y`. MySQL and SQLite only allow combining set operators with other set operators. So the only methods allowed after a set operator should be set operators in MySQL and SQLite. That should also simplify the SQL generation for these two dialects, since we'd only need to "join" the select queries with a selected set operator kind.

Reply:
Understood, I thought it would be easier maintaining it the the included instead of the excluded.
In MySql this is a totally valid query:
```sql
(select * from users order by id limit 3 offset 1) union (select * from users where id > 5 order by id limit 2 offset 1) limit 4 offset 1;
```
I will add the excluded methods for SQLite.

Reply:
Actually, order by, limit and offset are allowed in the "Compound Select Statements" (as they call it in SQLite) as long as it's in the end. But I don't think we have a way to enforce not allowing order by and limit on the selects via types. [See SQLite docs](https://www.sqlite.org/lang_select.html#compound_select_statements)

Reply:
I see, you are correct. Didn't find any info in MySQL docs about this syntax, but it indeed works. SQLite part is also correct.

Reply:
For the case of SQLite, what we can do is exclude all the set operators if a limit or order by is used in the left select. But I don't think we'll be able to enforce at the type level that the right select shouldn't have a limit or order by. Would you like me to throw a runtime error if limit or order by is detected in the right select?